src/pages/home.js

```diff
@@ -1,39 +1,37 @@
-import React from "react";
+import React, { useEffect } from "react";
 import CenteredTabs from "../components/CentredTabs";
 import NavBar from "../components/NavBar";
-import axios from "axios";
+import { connect } from 'react-redux';
+import { loadData } from '../actions';
 
-export default class HomePage extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            articles: [],
-            authors: [],
-            dataLoaded: false
-        }
-    }
-    componentWillMount() {
-        this.getData();
-    }
-    async getData() {
-        let articles = await axios.get("http://localhost:3001/articles");
-        let authors = await axios.get("http://localhost:3001/authors");
-        this.setState({ articles: articles, authors: authors, dataLoaded: true });
+function HomePage(props) {
+    const loadData = props.loadData;
+    useEffect((props) => {
+        const links = ["http://localhost:3001/articles", "http://localhost:3001/authors"]
+        loadData(links);
+    }, [loadData]);
+    const articlesLoaded = props.articles!==undefined && props.articles.length!==0;
```

Comment:
you can create some helper function like: `const isNotEmpty = (arr) => arr && arr.length !== 0;`
and use it like `const articlesLoaded = isNotEmpty(props.articles)`